### PR TITLE
Rename tezos nodes in various places

### LINKED ISSUE #352: Change "tezos-node" to name that uses the node's history mode
The current naming scheme of `tezos-node` doesn't make it obvious what the node is doing and being used for. We have more explicit naming in oxheadinfra, like `rolling-node`. We have to set `tezos-node: null` to override the default chart node values so that it isn't created. And then create the nodes we want, e.g. `archive-node`, `rolling-node`, etc.

 With this change, we won't have to worry about a `tezos-node` statefulset being created. However, if one doesn't want a rolling node, they would still need to set `rolling-node: null`.

### PR DESCRIPTION
**Pull request by harryttd**
closes #352

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,24 +138,16 @@ Connecting to a public net is easy!
...
 
-If you'd like to spin up a node with history mode full, run:
+Running this results in:
 
-```shell
-helm install tezos-mainnet oxheadalpha/tezos-chain \
---namespace oxheadalpha --create-namespace \
---set nodes.regular.tezos-node-0.config.shell.history_mode=full
-```
-
```
Comment (harryttd) [inline README.md:155]: I got rid of creating a full node and just left the user with creating the default rolling node. We would need to introduce needing a values.yaml here and referencing it, and I think it is not necessary at this point. We can't just `--set tezos-node....` anymore, bec the node names are not uniform. They reflect their history modes.

**Review by nicolasochem — CHANGES_REQUESTED** [review]
Looks good except some of your readme changes. I don't like `full-baking-node`, it sounds awkward. Plus the recommendation is to use rolling nodes for baking. Can you just name it `baking-node` and set it as rolling in the config example?

Also `full-tezedge-node` should be named `tezedge-full-node` I think.

**Review by nicolasochem — APPROVED** [review]
thanks!

**Comment by harryttd** [comment]
Thank you Sir

**Pull request merged**